docs/common/guides/jlink_image.adoc

```diff
@@ -0,0 +1,207 @@
+
+Now stop the running application (by pressing Ctrl+C).
+
+For more information about the Quickstart application and other enpoints it supports see the
+<<{helidon-lc-edition}/guides/02_quickstart.adoc,Helidon {helidon-uc-edition} quickstart Guide>>.
+
+== Building a Custom Runtime Image
+
+You can build a custom runtime image in 2 different ways:
+
+* Locally, on your desktop
+* Using Docker
+
+=== Local build
+
+Build the custom runtime image using the jlink image profile:
+
+[source,bash]
+----
+mvn package -Pjlink-image
+----
+
+[TIP]
+This uses the `helidon-maven-plugin` to perform the custom image generation.
+
+After the build completes it will report some statistics about the build including
+the reduction in image size.
+
+The `target/helidon-quickstart-{helidon-lc-edition}` directory is a self contained
+custom image of your application. It contains your application, its runtime
+dependencies and the JDK modules it depends on. You can start your application
+using the provide
+`start` script:
+
+[source,bash,subs="attributes+"]
+----
+./target/helidon-quickstart-{helidon-lc-edition}/bin/start
+----
+
+=== Class Data Sharing (CDS) Archive
+
+Aslo included in the custom image is a Class Data Sharing (CDS) archive that
+improves your application's startup performance and in-memory footprint.
+You can learn more about Class Data Sharing in the
+https://docs.oracle.com/en/java/javase/11/vm/class-data-sharing.html[JDK documentation].
+
+The CDS archive increases your image size to get these performance optimizations.
+It can be of significant size (tens of MB). The size of the CDS archive is
+reported at the end of the build output.
+
+If you rather have a smaller image size (with a slightly increased startup time) you
+can skip the creation of the CDS archive by executing your build like this:
+
+[source,bash]
+----
+mvn package -Pjlink-image -Djlink.image.addClassDataSharingArchive=false
+----
+
+For more information see the
+https://github.com/oracle/helidon-build-tools/tree/master/helidon-maven-plugin[`helidon-maven-plugin` documentation].
+
+=== Multi-stage Docker build
+
+To build a Docker image with a custom Java runtime image use the jlink
+Dockerfile included with the quickstart.
+
+[source,bash,subs="attributes+"]
+----
+docker build -t helidon-quickstart-{helidon-lc-edition}-jlink -f Dockerfile.jlink .
+----
+
+[TIP]
+This does a full build inside the Docker container. The first
+time you run it, it will take a while because it is downloading all
+of the Maven dependencies and caching them in a Docker layer.
+Subsequent builds will be much faster as long as you don't change
+the `pom.xml` file. If the pom is modified then the dependencies
+will be re-downloaded.
+
+Start the application:
+
+[source,bash,subs="attributes+"]
+----
+docker run --rm -p 8080:8080 helidon-quickstart-{helidon-lc-edition}-jlink:latest
+----
+
+You can exercise the application's endpoints as before.
```

Comment:
I think it would also be worth showing passing script arguments, e.g. `--test` or `--help` just so it is clear how it is done, (and maybe that the port mappings isn't required).